### PR TITLE
add pipeline support for Ascend NPU

### PR DESCRIPTION
**Pull request by ji-huazhong**
# What does this PR do?
Currently, Transformers's Trainer library has supported Ascend NPU([see](https://github.com/huggingface/transformers/pull/24879)).

This PR makes the pipeline library available on Ascend NPU. For example, we can run a text-classification pipeline on Ascend NPU with the following command(reference the example of `TextClassificationPipeline`):

```python
>>> from transformers import pipeline

>>> classifier = pipeline(model="distilbert-base-uncased-finetuned-sst-2-english", device="npu:1")

>>> classifier("Director tried too much.")
[{'label': 'NEGATIVE', 'score': 0.9963733553886414}]
```

## Who can review?
Anyone in the community is free to review the PR once the tests have passed.

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -89,6 +90,9 @@
 XLA_USE_BF16 = os.environ.get("XLA_USE_BF16", "0").upper()
 XLA_DOWNCAST_BF16 = os.environ.get("XLA_DOWNCAST_BF16", "0").upper()
 
+if is_torch_npu_available():
+    import torch_npu  # noqa: F401
```
Comment (statelesshz) [inline src/transformers/modeling_utils.py:94]: We need it to register the NPU backend for PyTorch.

File: src/transformers/pipelines/__init__.py [hunk 2]
```diff
@@ -107,6 +108,9 @@
 if is_torch_available():
     import torch
 
+    if is_torch_npu_available():
+        import torch_npu  # noqa: F401
```
Comment (statelesshz) [inline src/transformers/pipelines/__init__.py:112]: as above

**Comment by statelesshz** [comment]
@Narsil Good day. Could you please take a look at this PR? Thanks :-)
BTW, it seems that the failing test case is not related to this PR.

**Comment by statelesshz** [comment]
rebase commits from the main branch.

**Comment by statelesshz** [comment]
@muellerzr and @Narsil Good day. Please take a look at this PR. This PR is mainly to register the NPU backend in the pipeline, and then we can pass `device=npu` to the `pipeline` method

**Comment by ArthurZucker** [comment]
Hey! Sorry I don´t think this is the way we want to proceed to support other hardwares cc @Narsil I think you were thinking of a more general way of doing this no?

**Comment by Narsil** [comment]
I think if users send directly `torch.device("npu")`  it should actually work.
Since torch added support for many more hardware, I think it's gonna be hard to add such support for all of them

**Comment by muellerzr** [comment]
@Narsil could we not just make use of the `PartialState().device` from Accelerate and let us handle that for you? Even if you don't use any other aspect, it can pick up the device and just be used for that sole purpose :)

**Pull request closed without merging**